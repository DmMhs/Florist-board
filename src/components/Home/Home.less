.Home {
  width: 100%;
  min-height: calc(100vh - 9rem);
  hr {
    width: 50%;
    border: 0.5px solid #eee;
  }
  .slider-wrapper {
    width: 60%;
    height: 400px;
    margin: 0 auto;
    .Slider {
      width: 100%;
      height: 100%;
      margin-top: 2rem;
      .Slide {
        background-position: bottom;
      }
    }
  }
  @media screen and (max-width: 768px) {
    .Spinner {
      position: absolute;
      top: 50%;
      bottom: 50%;
      left: 50%;
      right: 50%;
    }
    .slider-wrapper {
      position: fixed;
      height: 100%;
      width: 100%;
      .Slider {
        width: 100%;
        height: 100%;
        .Slide {
          background-position: center center;
        }
      }
    }
  }
}
