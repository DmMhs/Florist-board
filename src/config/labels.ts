const labels = {
  brand: 'FLORIST.UA',
  adress: 'MARKET SQUARE, LVIV, UA',
  postCode: '346346',
  phone: '+123-45-678',
  navigation: {
    home: 'HOME',
    shop: 'SHOP',
    gallery: 'GALLERY',
    contacts: 'CONTACTS',
    cart: 'CART',
    account: {
      main: 'MY ACCOUNT',
      menu: {
        signIn: 'SIGN IN',
        signUp: 'SIGN UP'
      }
    }
  }
};

export default labels;
