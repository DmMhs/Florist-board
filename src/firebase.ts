import * as firebase from 'firebase';

export const config = {
  apiKey: 'AIzaSyB7esiw1XYzS2_hllELqqYzqN5wIQav0Oc',
  authDomain: 'florist-cb933.firebaseapp.com',
  databaseURL: 'https://florist-cb933.firebaseio.com',
  projectId: 'florist-cb933',
  storageBucket: 'florist-cb933.appspot.com',
  messagingSenderId: '582977319235'
};

firebase.initializeApp(config);
export const database = firebase.database();
export const homeImagesRef = database.ref('home-images');
export const productsRef = database.ref('products');
export const galleryImagesRef = database.ref('gallery-images');
