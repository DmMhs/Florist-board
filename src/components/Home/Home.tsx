import React, { Component } from 'react';
import Slider from '../Slider/Slider';

import './Home.less';

class Home extends Component {
  render() {
    return (
      <div className="Home">
        <Slider
          images={[
            'assets/images/florist-cut.png',
            'assets/images/Florist-banner-cutted.jpg',
            'assets/images/florist-cut.png',
            'assets/images/Florist-banner-cutted.jpg'
          ]}
        />
      </div>
    );
  }
}

export default Home;
