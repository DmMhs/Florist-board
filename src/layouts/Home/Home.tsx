import React, { Component } from 'react';

import { homeImagesRef, database } from '../../firebase';
import Spinner from '../../components/Spinner/Spinner';
import Slider from '../../components/Slider/Slider';
import './Home.less';
import { AppContext } from '../../AppContext';

interface HomeProps {}

interface HomeState {
  bannerImages: string[];
  isFetching: boolean;
  sliderWrapperWidth: string;
  sliderWrapperHeight: string;
}

let resizeListener: EventListener;

class Home extends Component<HomeProps, HomeState> {
  constructor(props: HomeProps) {
    super(props);
    this.state = {
      bannerImages: [],
      isFetching: false,
      sliderWrapperHeight: '',
      sliderWrapperWidth: ''
    };
  }

  public componentDidMount() {
    homeImagesRef.on('value', snapshot => {
      this.setState({
        isFetching: true
      });
      const newImages: string[] = [];
      snapshot!.forEach(imgRef => {
        newImages.push(imgRef.val());
      });
      this.setState({
        bannerImages: newImages
      });
      database.ref('banner-params').on('value', snapshot2 => {
        this.setState({
          sliderWrapperHeight: snapshot2!.val().desktopBannerWrapperHeight,
          sliderWrapperWidth: snapshot2!.val().desktopBannerWrapperWidth,
          isFetching: false
        });
      });
    });
    resizeListener = () => {
      this.setState({
        sliderWrapperHeight: '',
        sliderWrapperWidth: ''
      });
    };

    window.addEventListener('resize', resizeListener);
  }

  public componentWillUnmount() {
    window.removeEventListener('resize', resizeListener);
  }

  public render() {
    const {
      bannerImages,
      sliderWrapperWidth,
      sliderWrapperHeight
    } = this.state;
    const context = this.context;
    const mobileModeEnabled = context.state.mobileMode;
    const bannerStyle =
      mobileModeEnabled === true
        ? {}
        : {
            width: sliderWrapperWidth,
            height: sliderWrapperHeight
          };
    return (
      <div className="Home">
        {bannerImages.length <= 0 ? <Spinner /> : null}
        <div className="slider-wrapper" style={bannerStyle}>
          <Slider
            images={bannerImages}
            auto={true}
            showControls={true}
            bannerModeEnabled={true}
          />
        </div>
      </div>
    );
  }
}
Home.contextType = AppContext;
export default Home;
