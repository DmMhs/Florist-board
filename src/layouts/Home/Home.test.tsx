import React from 'react';
import { shallow, mount } from 'enzyme';

import { homeImagesRef } from '../../firebase';
import Home from './Home';
import { BrowserRouter } from 'react-router-dom';
import AppContextProvider from '../../AppContext';
import labels from '../../config/labels';

describe('Home works as expected', () => {
  it('Home component matches a snapshot', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <Home />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('has a valid state value', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <Home />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      fetchInProgress: false,
      mobileMode: false
    });
    wrapper.update();
    const instance = wrapper.instance();
    instance.setState({
      bannerImages: ['zero', 'one', 'two']
    });

    expect(instance.state.bannerImages.length).toEqual(3);
  });

  it('banner images do fetch', async () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <Home />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      fetchInProgress: false,
      mobileMode: false
    });
    wrapper.update();
    const instance = wrapper.instance();

    await homeImagesRef.once('value').then(snapshot => {
      instance.setState({
        isFetching: true
      });
      const newImages: string[] = [];
      snapshot!.forEach(imgRef => {
        newImages.push(imgRef.val());
      });
      instance.setState({
        bannerImages: newImages,
        isFetching: false
      });
    });
    expect(instance.state.bannerImages.length).toBeGreaterThan(0);
  });
});
