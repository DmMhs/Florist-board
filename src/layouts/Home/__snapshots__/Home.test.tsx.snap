// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home works as expected Home component matches a snapshot 1`] = `
<div
  className="Home"
>
  <WithStyles(CircularIndeterminate) />
  <div
    className="slider-wrapper"
  >
    <Slider
      auto={true}
      bannerModeEnabled={true}
      images={Array []}
      showControls={true}
    />
  </div>
</div>
`;
